nixos/modules/services/misc/matrix-registration.nix

```diff
@@ -0,0 +1,218 @@
+            type = types.str;
+            description = "The server_name";
+            example = "example.tld";
+            default = optionalString config.services.matrix-synapse.enable
+              config.services.matrix-synapse.server_name;
+          };
+          registration_shared_secret = mkOption {
+            type = types.str;
+            description =
+              "The registration shared secret with the synapse configuration";
+            example = "RegistrationSharedSecret";
+            default = optionalString config.services.matrix-synapse.enable
+              config.services.matrix-synapse.registration_shared_secret;
+          };
+          admin_secret = mkOption {
+            type = types.str;
+            description = "The admin secret to access the API";
+            example = "APIAdminPassword";
+          };
+          base_url = mkOption {
+            type = types.str;
+            description = "A prefix for all endpoints";
+            default = "";
+          };
+          client_redirect = mkOption {
+            type = types.str;
+            description =
+              "The client instance to redirect after succesful registration";
+            default = "https://app.element.io/#/login";
+          };
+          client_logo = mkOption {
+            type = types.path;
+            description = "The client logo to display";
+            default =
+              "${pkgs.matrix-registration}/lib/python3.8/site-packages/matrix_registration/static/images/element-logo.png";
+          };
+          db = mkOption {
+            type = types.str;
+            default = "sqlite:///${dataDir}/db.sqlite3";
+            description = "Where to store the database";
+          };
+          host = mkOption {
+            type = types.str;
+            default = "localhost";
+            description = "What host to listen on";
+          };
+          port = mkOption {
+            type = types.port;
+            default = 5000;
+            description = "What port to listen on";
+          };
+          rate_limit = mkOption {
+            type = types.listOf types.str;
+            default = [ "100 per day" "10 per minute" ];
+            description =
+              "How often is one IP allowed to access matrix-registration";
+          };
+          allow_cors = mkOption {
+            type = types.bool;
+            default = false;
+            description = "Allow Cross Origin Resource Sharing";
+          };
+          ip_logging = mkOption {
+            type = types.bool;
+            default = false;
+            description = "Save IPs in the database";
+          };
+          logging = mkOption {
+            type = types.attrs;
+            description = "Python logging configuration";
+            default = {
+              disable_existing_loggers = false;
+              version = 1;
+              root = {
+                level = "DEBUG";
+                handlers = [ "console" "file" ];
```

Comment:
Do we need 2 loggers? Generally we try to stick to `journald` logging (in this case via `stdout`).

Reply:
No not necessary, I'm going to remove `file`